addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -397,6 +398,7 @@ export class LinkPlugin extends Plugin {
                     } else {
                         this.linkElement.removeAttribute("class");
                     }
+                    cleanTrailingBR(closestBlock(this.linkElement));
```

Comment:
I agree here that by removing the `br` here would eliminate the issue of applying styles. But what about the linebreak `br`. Let's say you have `<p><a>[abc.com</a><br>]<br></p>` => apply font size => `<p><a><span class="someFontSizeClass">[abc.com</span></a><span class="someFontSizeClass"><br></span>]<br></p>` => now select the text => `<p><a><span class="someFontSizeClass">[abc.com]</span></a><span class="someFontSizeClass"><br></span><br></p>` => removeFormat => and it will result in `<p><a>[abc.com]</a><span class="someFontSize"><br></span><br></p>` and the bug will still occur. I think this will also need a fix in `formatSelection` where it should filter out the linebreak `br`, specifically those representing line breaks, not the ones indicating empty lines.

Reply:
The case you mentioned should indeed not format the `<br>` but since [this](https://github.com/odoo/odoo/commit/df1dcf1c6bfc94ae8e084b41c1b5cc84b2cc5203) commit has merged, all the selected `<br>`s will get formatted. Please note that we can't filter out the line-break `<br>`s and format only the ones indicating empty lines because we will have problem if we are dealing with multiple line breaks: `<p>test[<br><br><br><br><br>]</p>` .
Right now it works fine by wrapping all the `br`s in a single `span` . However, if the commit I mentioned only addresses the issue related to table (as mentioned in the commit message) then we can fix that commit and the format will not be applied to any `br` (as it was before).